tools/buildsys/src/deps.rs

```diff
@@ -0,0 +1,196 @@
+    let something = query.resolve_with_fn(|query, link| {
+        let to = link.to();
+        is_manifest_type(&to, ManifestType::Package)
+    });
+    let mut packages: Vec<String> = something
+        .packages(DependencyDirection::Forward)
+        .filter_map(|package_metadata| {
+            // The first dependency is always included, and if it happens to be something other than
+            // a package, we want to filter it out.
+            if is_manifest_type(&package_metadata, ManifestType::Package) {
+                // Return the name instead of the metadata or package ID.
+                Some(package_metadata.name().to_string())
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    // Sort so that this function has consistent, dependable output regardless of graph internals.
+    packages.sort();
+    Ok(packages)
+}
+
+// TODO - move this to the Manifest struct
+/// List all kits needed to build a package or variant.
+pub fn kit_dependencies(manifest_path: impl AsRef<Path>) -> Result<Vec<String>> {
+    let manifest_path = manifest_path.as_ref();
+    // TODO - deserialize instead of using a MetadataCommand.
+    let mut kits: Vec<String> = MetadataCommand::new()
+        .manifest_path(&manifest_path)
+        .build_graph()
+        .unwrap()
+        .packages()
+        .filter_map(|package_metadata| {
+            if is_manifest_type(&package_metadata, ManifestType::Kit) {
+                Some(package_metadata.name().to_string())
+            } else {
+                None
+            }
+        })
+        .collect();
+    kits.sort();
+    Ok(kits)
+}
+
+fn is_manifest_type(package_metadata: &PackageMetadata, manifest_type: ManifestType) -> bool {
+    let metadata_table = package_metadata.metadata_table();
+    match manifest_type {
+        ManifestType::Package => metadata_table.get("build-package").is_some(),
+        ManifestType::Kit => metadata_table.get("build-kit").is_some(),
+    }
+}
+
+fn find_id(name: &str, graph: &PackageGraph, manifest_type: ManifestType) -> Option<PackageId> {
+    for package_metadata in graph.packages() {
+        if is_manifest_type(&package_metadata, manifest_type) && package_metadata.name() == name {
+            return Some(package_metadata.id().to_owned());
+        }
+    }
+    None
+}
+
+/// Because the PackageGraph uses PackagedID, and it is not easy for us to calculate the PackageID
+/// ourselves, we need to know the name of our manifest so that we can look up its PackageID in the
+/// graph. The simplest way to do that seems to be to deserialize into TOML and look for the
+/// `package.name` key.
+fn get_name_and_type(manifest_path: impl AsRef<Path>) -> Result<(String, ManifestType)> {
+    let path = manifest_path.as_ref();
+    let contents = fs::read_to_string(&path).unwrap();
+    let table: Table = toml::from_str(&contents).unwrap();
+    let package_table = table.get("package").unwrap().as_table().unwrap();
```

Comment:
You could turn this into a `ManifestInfo` - probably we should have a helper method there that returns the preferred package name based on the crate name or metadata override, and another helper to return the type.

Reply:
Kinda ended up doing this although I introduced another type named `Manifest` that includes the graph, and left `ManifestInfo` unchanged. (This is because `pubsys` and `bottlerocket-variant` are using `ManifestInfo` in a way where the graph isn't needed.)